lib/roby/relations/dependency.rb

```diff
@@ -921,7 +921,7 @@ def pretty_print(pp) # :nodoc:
             if mode == :failed_event
                 pp.text "triggered the failure predicate '#{relation[:failure]}': "
             elsif mode == :unreachable_success
-                pp.text "cannot reach the success condition '#{relation[:success]}': "
+                pp.text "child terminated, success condition can no longer be reached '#{relation[:success]}': "
```

Comment:
"child terminated" is wrong here. The success condition can be anything, and can fail for other reasons than the task being finished. I like the rephrasing to "success condition can no longer be reached" though.